internal/update/update.go

```diff
@@ -0,0 +1,151 @@
+package update
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/charmbracelet/crush/internal/version"
+)
+
+const (
+	githubAPIURL = "https://api.github.com/repos/charmbracelet/crush/releases/latest"
+	userAgent    = "crush-update-check"
+)
+
+// Release represents a GitHub release.
+type Release struct {
+	TagName string `json:"tag_name"`
+	HTMLURL string `json:"html_url"`
+}
+
+// UpdateInfo contains information about an available update.
+type UpdateInfo struct {
+	CurrentVersion string
+	LatestVersion  string
+	ReleaseURL     string
+	Available      bool
+}
+
+// CheckForUpdate checks if a new version is available.
+func CheckForUpdate(ctx context.Context) (*UpdateInfo, error) {
+	info := &UpdateInfo{
+		CurrentVersion: version.Version,
+	}
+
+	// Skip update check for development versions.
+	if strings.Contains(version.Version, "unknown") {
+		return info, nil
+	}
+
+	release, err := fetchLatestRelease(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to fetch latest release: %w", err)
+	}
+
+	info.LatestVersion = strings.TrimPrefix(release.TagName, "v")
+	info.ReleaseURL = release.HTMLURL
+
+	// Compare versions.
+	if compareVersions(info.CurrentVersion, info.LatestVersion) < 0 {
+		info.Available = true
+	}
+
+	return info, nil
+}
+
+// fetchLatestRelease fetches the latest release information from GitHub.
+func fetchLatestRelease(ctx context.Context) (*Release, error) {
+	client := &http.Client{
+		Timeout: 30 * time.Second,
+	}
+
+	req, err := http.NewRequestWithContext(ctx, "GET", githubAPIURL, nil)
+	if err != nil {
+		return nil, err
+	}
+	req.Header.Set("User-Agent", userAgent)
+	req.Header.Set("Accept", "application/vnd.github.v3+json")
+
+	resp, err := client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		body, _ := io.ReadAll(resp.Body)
+		return nil, fmt.Errorf("GitHub API returned status %d: %s", resp.StatusCode, string(body))
+	}
+
+	var release Release
+	if err := json.NewDecoder(resp.Body).Decode(&release); err != nil {
+		return nil, err
+	}
+
+	return &release, nil
+}
+
+// compareVersions compares two semantic versions.
+// Returns -1 if v1 < v2, 0 if v1 == v2, 1 if v1 > v2.
+func compareVersions(v1, v2 string) int {
```

Comment:
probably better to use Masterminds/semver

Reply:
ah, make sense will update